examples/fungible-token/tests/sim/utils.rs

```diff
@@ -19,10 +20,10 @@ const DEFI_ID: &str = "defi";
 // Register the given `user` with FT contract
 pub fn register_user(user: &near_sdk_sim::UserAccount) {
     user.call(
-        FT_ID.to_string(),
+        FT_ID.parse().unwrap(),
```

Comment:
what would be the benefit to using that over `AccountId::new_unchecked`? How would the extra bit be relevant? Seems clunky to have both so we should think through what's best long term.

Reply:
Sorry for channeling  unfiltered stream of consciousness :)

From looking at the usages here, it seems that "hard-coded" account names are a somewhat frequent use-case, at least for tests. So, it would be more user-friendly if the API allowed to directly express constant accounts, such that runtime validation and unwrapping is not required. 

This is doable, we can have a String-like type which is both const-constructable and supports dynamic allocation, the simplest version would be just `Cow<'static, str>`. The cow would add some extra space overhead though. 

It is possible to implement a string type which has the same size_of as `std::String`, but which is const-constructable and avoids allocation for short strings. https://github.com/rust-analyzer/smol_str/ is a half-baked attempt at that, it can be improved quite a bit.

Reply:
Seems like a bit of a micro-optimization that might be out of the scope of this PR. I like the idea but should probably have better infrastructure for testing contract size and performance before trying to eagerly optimize.

As for skipping runtime validation, that's what `AccoundId::new_unchecked` currently services. Is there a better API for this? I see some need to have a runtime string not be validated, so having two methods would be confusing and would limit us from switching the internal type

Reply:
Yeah, this is def out of scope, something for the time when all the base APIs are done and future-proofed.